packages/react/src/components/modal/types.ts

```diff
@@ -0,0 +1,78 @@
+import { CSSProperties, ReactNode } from 'react';
+import { DeviceContextProps } from '../device-context-provider/device-context-provider';
+
+export interface StyledModalProps extends Pick<DeviceContextProps, 'breakpoints' | 'isMobile'> {
+    $width: CSSProperties['width'];
+}
+
+export interface ContentProps extends Pick<DeviceContextProps, 'isMobile'> {
+    $noPadding: boolean;
+    $hasHeader?: boolean;
+    $hasFooter?: boolean;
+}
+
+export interface HeaderProps extends ContentProps {
+    $hasCloseButton: boolean;
+    $isTopScrolled?: boolean;
+}
+
+export interface FooterProps extends ContentProps {
+    $isBottomScrolled?: boolean;
+}
+
+export type MobileDeviceContextProps = Pick<DeviceContextProps, 'isMobile'>
+
+export interface BaseModalProps {
+    /** Takes a query selector targeting the app Element. */
+    appElement?: string;
+    ariaDescribedby?: string;
+    /** Boolean indicating if the appElement should be hidden. Defaults to true.
+     * Should only be used for test purposes. */
+    ariaHideApp?: boolean;
+    children?: ReactNode;
+    className?: string;
+    /**
+     * Modify the modal width
+     * @default 60vw
+     */
+    width?: CSSProperties['width'];
+    /**
+     * Adds "x" iconButton to close modal
+     * @default true
+     */
+    hasCloseButton?: boolean;
+    isOpen: boolean;
+    /**
+     * Defines if the overlay click should close the modal
+     * @default true
+     */
+    shouldCloseOnOverlayClick?: boolean;
+    parentSelector?: () => HTMLElement;
```

Comment:
L'explication est t'il nécessaire ? Selon moi le code devrait être assez explicite pour qu'on puisse savoir ce que c'est props fait 🤔 .
De plus, le `Should only be used for test purposes.`. J'ignore si c'est vraiment la seule manière de le tester, mais je n'aime pas trop l'idée de créer un props juste pour les tests. 
Personnellement, il y a un truc d'agaçant ici, mais rien de problèmatique si c'est la meilleure solution dans cette situation.